lib/fog/vsphere/requests/compute/list_child_snapshots.rb

```diff
@@ -0,0 +1,68 @@
+module Fog
+  module Compute
+    class Vsphere
+      class Real
+        def list_child_snapshots(snapshot, opts = {})
+          s = Hash === snapshot ?
+            Snapshot.new(snapshot.update(service: self)) : snapshot
+          s.tree_node.childSnapshotList.map do |snap|
+            item = child_snapshot_info(snap, s)
+            [
+              item,
+              opts[:recursive] ? list_child_snapshots(item, opts) : nil
+            ]
+          end.flatten.compact
```

Comment:
Could you assign the result of `s.tree_node.childSnapshotList.map` to a variable and then call `.flatten.compact` on that variable at the end of your method?
Chaining methods to block like this isn't that readable nor easy to follow.